api/discovery/v1alpha1/foreigncluster_types.go

```diff
@@ -37,27 +37,35 @@ const (
 	IncomingPeeringDiscovery DiscoveryType = "IncomingPeering"
 )
 
+type TrustMode string
+
+const (
+	TrustUnknown   TrustMode = ""
```

Comment:
Just a personal preference: I would name these constants differently (for example  `TrustModeUnknown` or `ModeUnknown` or simply `Unknown`, depending on how you name the respective field in the struct) so that when you compare it is clearer the meaning.